tedge/src/cli/connect/bridge_config.rs

```diff
@@ -0,0 +1,356 @@
+use crate::cli::connect::ConnectError;
+use tedge_config::FilePath;
+use url::Url;
+
+#[derive(Debug, PartialEq)]
+pub struct BridgeConfig {
+    pub cloud_name: String,
+    pub config_file: String,
+    pub connection: String,
+    pub address: String,
+    pub remote_username: Option<String>,
+    pub bridge_root_cert_path: FilePath,
+    pub remote_clientid: String,
+    pub local_clientid: String,
+    pub bridge_certfile: FilePath,
+    pub bridge_keyfile: FilePath,
+    pub use_mapper: bool,
+    pub try_private: bool,
+    pub start_type: String,
+    pub clean_session: bool,
+    pub notifications: bool,
+    pub bridge_attempt_unsubscribe: bool,
+    pub topics: Vec<String>,
+}
```

Comment:
As we are using strong types for the `BridgeConfigAzureParams` and `BridgeConfigC8yParams` shouldn't we also use strong types here?
Eg:
* pub address: String, -> pub address: Url,
* pub config_file: String -> pub config_file: PathBuf / Buf / FilePath
* pub cloud_name: String ->  pub cloud_name: Cloud

Reply:
Very good point! We definitively should!

One issue I came across when trying to convert `address` to an `Url` was that the way we use `Url` is slightly wrong. See 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d9b19f34c756778ddd86ce026c874a9e

When parsing `test.test.io:8883`, we actually treat `test.test.io` as schema and `8883` as path. 

About `config_file`... that's a good point, too. This is used as the name of the config file, not the full path to it. I was concerned that changing that to a `FilePath` etc. people might confuse it with the full path.

`Cloud_name`. Yes, you are right again. 

We should add more precise types.

Reply:
@makr11st I'd be much much better if we could move the `Url::parse` to where we create the `BridgeConfig`. We still need the validation, but that would only check if the files actually exist. As it isn't that simple as it seems, I'd like to keep that for a possible future task :)

Reply:
I think I exactly know what you mean as I was dealing with the same problem, `Url::parse` is correct here as it adheres to relevant url spec: https://url.spec.whatwg.org/ .